torch/utils/checkpoint.py

```diff
@@ -194,7 +194,7 @@ def set_device_states(devices, states) -> None:
 
 
 def _get_autocast_kwargs(device="cuda"):
-    if _supports_autocast(device):
+    if torch._C._is_autocast_available(device):
```

Comment:
In addition, I have a quick question. Is it reasonable to add torch.xpu.amp.autocast which is equivalent to torch.autocast('xpu')? PyTorch assumes that each backend has its own autocast in many places, like
https://github.com/pytorch/pytorch/blob/34bce27f0d12bf7226b37dfe365660aad456701a/torch/utils/checkpoint.py#L301

If not, xpu will be not supported in torch.xxx.amp.autocast scenario. Note torch.cuda.amp.autocast is not equivalent to torch.autocast('cuda') as the former one can handle JIT path. If use torch.autocast(device_type), it can not handle CUDA/CPU JIT path.

Reply:
The answer here is a bit tricky. The very honest answer is:
- If you really want to do it, I understand you might need TorchScript (TS) support and I'm not going to block it on principle.
- But think most of our users don't care about that, I personally don't know about TS enough and I have limited bw to review PRs. So I don't personally want to sign up to review this.
- So you need to find someone to review your TS-related. Given that there are 0 engineers from the compiler team working on TS, I have no idea who to point you to to help with this.
- I unfortunately don't know what is good solution to recommend from here. If you don't have any user hard-blocked on this, I would suggest dropping it as the simplest thing to do. If you have users that don't have any other option but using TS, you most likely will need to escalate this through the compiler team.

Reply:
We don't like to support TorchScript on XPU. We just intend to add `torch.xpu.amp.autocast`, a wrapper of `torch.amp.autocast` specific to `xpu`, to support `torch.xxx.amp.autocast` code-style scenario on eager mode, like the below code defined in `torch.xpu.amp.autocast_mode.py`
```python
class autocast(torch.amp.autocast_mode.autocast):
    r"""
    See :class:`torch.autocast`.
    ``torch.xpu.amp.autocast(args...)`` is equivalent to ``torch.autocast("xpu", args...)``
    """

    def __init__(self, enabled=True, dtype=torch.bfloat16, cache_enabled=True):
        super().__init__(
            "xpu", enabled=enabled, dtype=dtype, cache_enabled=cache_enabled
        )
```

Reply:
I'm not sure what is the benefit of such API? We are adding generic API so that users don't have to use these one-off APIs that make their code locked onto a specific device.

Reply:
@albanD 
This is a history issue, indeed. Before this PR, user has been used to adopting torch.cuda.amp.autocast in their scripts.
We followed CUDA to provide torch.xpu.amp.autocast as a public API in our extension.
Although we have torch.amp.autocast now, it's difficult to ask all users to replace with new API immediately.

Reply:
I'm not sure to follow.
The existing code contains torch.cuda.XXX calls. So user code must be changed to support xpu. Adding a new torch.xpu.XXX API doesn't remove the need for user-code change.
And if we're changing user code, we might as well change it to use torch.amp.XXX API directly.
Am I missing something here?